src/Arduino_LoRaConnectionHandler.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+   This file is part of ArduinoIoTCloud.
+
+   Copyright 2019 ARDUINO SA (http://www.arduino.cc/)
+
+   This software is released under the GNU General Public License version 3,
+   which covers the main part of arduino-cli.
+   The terms of this license can be found at:
+   https://www.gnu.org/licenses/gpl-3.0.en.html
+
+   You can be released from the requirements of the above licenses by purchasing
+   a commercial license. Buying such a license is mandatory if you want to modify or
+   otherwise use the software for commercial activities involving the Arduino
+   software without disclosing the source code of your own applications. To purchase
+   a commercial license, send an email to license@arduino.cc.
+*/
+
+/******************************************************************************
+   INCLUDE
+ ******************************************************************************/
+
+#if defined(ARDUINO_SAMD_MKRWAN1300) || defined(ARDUINO_SAMD_MKRWAN1310) /* Only compile if the board has LoRa */
+
+#include "Arduino_LoRaConnectionHandler.h"
+
+/******************************************************************************
+   CONSTANTS
+ ******************************************************************************/
+
+static const unsigned long NETWORK_CONNECTION_INTERVAL = 30000;   /*    NOT USED    */
+
+/******************************************************************************
+   CTOR/DTOR
+ ******************************************************************************/
+
+LoRaConnectionHandler::LoRaConnectionHandler(const char *_appeui, const char *_appkey, _lora_band band) :
+  appeui(_appeui),
+  appkey(_appkey),
+  band(band),
+  lastConnectionTickTime(millis()),
+  connectionTickTimeInterval(CHECK_INTERVAL_IDLE),
+  keepAlive(false),
+  _on_connect_event_callback(NULL),
+  _on_disconnect_event_callback(NULL),
+  _on_error_event_callback(NULL) {
+  netConnectionState = NetworkConnectionState::INIT;
+}
+
+/******************************************************************************
+   PUBLIC MEMBER FUNCTIONS
+ ******************************************************************************/
+
+void LoRaConnectionHandler::init() {
+}
+
+// INIT, CONNECTING, CONNECTED, DISCONNECTING, DISCONNECTED, CLOSED, ERROR
+void LoRaConnectionHandler::addCallback(NetworkConnectionEvent const event, OnNetworkEventCallback callback) {
```

Comment:
Since this method is available for every type of connection handler it can be pulled up in the base class `ConnectionHandler` . Otherwise we have code duplication with `Arduino_TcpIpConnectionHandler`.

Reply:
Refactored